### DISCUSSION TITLE
problems in useing joints in sympy.physics.mechanics

### DISCUSSION THREAD

**Discussion by stein104** [sympy/sympy#25613] [Q&A] [answered]
I'm trying to understand the joints framework within sympy.physics.mechanics to define mechanical problems and generate the equation of motion. Im using the [Multi Degree of Freedom Holonomic System](https://docs.sympy.org/latest/modules/physics/mechanics/examples/multi_degree_freedom_holonomic_system.html). as a reference.
I have defined 2 examples with springs and a rod.  
My problem is that in both examples, for no reason that I can think of, the equation of motion is increased by  +"0.25*m*l^2".
compared to what I should get as EOM. 
I would really appreciate if someone could tell me were the "0.25*m*l^2" comes from and if my approach defining the examples is good. (Im quite new to python and coding in general) 

Example1:
![image](https://github.com/sympy/sympy/assets/143089361/42682190-9662-4820-9ded-ff294b0aad52)

```py
from sympy import symbols
from sympy.physics.mechanics import Body, PinJoint, WeldJoint, JointsMethod, inertia, dynamicsymbols
### gen. Koord. und Symbole erstellen
q1, u1 =dynamicsymbols('phi, phi.')
l, g, k, Izz = symbols('l, g, c, Izz') 
m = symbols('m')
### Körper erstellen:
wall1 = Body('N1')  
#Izz=m*l**2 
IB = inertia(wall1.frame, 0, 0, Izz)
rod = Body('R', mass=m, central_inertia=IB)
point_mass = Body('P', mass=m/4)
bodies = (wall1, rod, point_mass)
### create joints
rev = PinJoint('J', wall1, rod, coordinates=q1, speeds=u1, 
                joint_axis=wall1.z,
                child_point=l/2*rod.x) 
fix = WeldJoint('K', rod, point_mass, parent_point=l*(3/2)*rod.frame.x)   
joints = (rev, fix)
### add Forces/loads
left_end = rod.masscenter.locatenew('left_end', -l*(3/2) * rod.frame.x)
#spring forces
rod.apply_force(-k*(q1*2*l)*rod.y, point=left_end) 
rod.apply_force(-k*(q1*2*l)*rod.y, point=left_end) 
### calc EOM 
method = JointsMethod(wall1.frame, rev, fix) 
method.form_eoms() 
```
output:
![image](https://github.com/sympy/sympy/assets/143089361/02ec0e99-5827-470e-93e1-1908b4670406)

Example 2:
![image](https://github.com/sympy/sympy/assets/143089361/7c04a0e7-cb65-49a0-bc2f-37c8c068c4a5)
```py
from sympy import*
from sympy.physics.mechanics import*
### gen. Coord
q1, u1 = dynamicsymbols('phi, phi.')
l, m, c, g, Izz, k =symbols('l, m, c, g, Izz, k')
#Izz = (1/3)*m*l**2
### bodies
wall1 = Body('A')
I = inertia(wall1.frame, 0, 0, Izz) 
stab = Body('C', mass=m, central_inertia=I)
bodies = (stab, wall1)
### joints
rev1 = PinJoint('J1', wall1, stab, coordinates=q1, speeds=u1,
                joint_axis=wall1.z, child_point=-l*1/2*stab.x) 
### forces
right_end = stab.masscenter.locatenew('right_end', (l/2) * stab.frame.x) 
stab.apply_force(m*g/2*stab.y, point=right_end)    #spring preload
stab.apply_force(k*(sin(q1)*l)*stab.y, point=right_end) 
stab.apply_force(-wall1.y*stab.mass*g) 
### EOM
method = JointsMethod(wall1, rev1)
method.form_eoms()  
```

output:
![image](https://github.com/sympy/sympy/assets/143089361/05c3702d-a819-4d57-a1ae-1ed002997cde)

**Comment by moorepants** [comment, marked as answer]
This makes a pin joint a distance of l/2 from the rod's mass center:
```
rev = PinJoint('J', wall1, rod, coordinates=q1, speeds=u1, 
                joint_axis=wall1.z,
                child_point=l/2*rod.x)
```

This glues the point mass to l*3/2 away from the rod's mass center.

``` 
fix = WeldJoint('K', rod, point_mass, parent_point=l*(3/2)*rod.frame.x)   
joints = (rev, fix)
```

```
left_end = rod.masscenter.locatenew('left_end', -l*(3/2) * rod.frame.x)
```

So it seems your rod is `3*l` long and the mass center of the rod is located at the rod's geometric center but the pin joint is located `l/2` from its mass center. And then point mass is `l*3/2` from the rod's mass center.

The total inertia of this rod + mass will then be:

``Izz + (m/4)*(3*l/2)**2 - (m/4)*(l/2)**2``

which is:

``Izz + 0.5*m*l**2``

**Reply by stein104** [reply]
Hi, I appreciate you answering me.
can you please explain to me how you get the Therm ```(m/4)*(3*l/2)**2 - (m/4)*(l/2)**2``` for the inertia of the point Mass m/4 around the pin joint? for me, it is just ```(m/4)*l**2```. Why do you need to calculate the inertia of the point mass around the canter of the rod and from there to the pin joint?
Also, my textbook calculates it this Way and deviates from what the Joints method gives me.

![image](https://github.com/sympy/sympy/assets/143089361/034d05a8-3af4-4646-bfb3-aad9bb3f3f45)

**Reply by moorepants** [reply]
Your pivot is not at the mass center so you need to use the parallel axis theorem to get the additional term.

**Reply by moorepants** [reply]
If you have a rod and a point mass that make up a composite rigid body, then you need to combine their inertia. Inertias can only be summed if the parallel axis theorem is applied.

**Reply by stein104** [reply]
Ok thanks. so the solution for the inertia here is wrong ?
![image](https://github.com/sympy/sympy/assets/143089361/9ed99afe-25b3-4881-a23d-6572ad8fbd92)

**Reply by moorepants** [reply]
I don't know, I haven't really tried to asses this textbook item. I don't know the assumptions they are making in the textbook.
